src/User.php

```diff
@@ -0,0 +1,27 @@
+<?php
+namespace Pokedex;
+
+
+class User
+{
+    private $email;
+
+    /**
+     * User constructor that checks if email address exists in the database.  Adds email address if it does not exist.
+     * @param AbstractEmail $email the validated email address from the user
+     * @param resource $db the database connection
+     */
+    public function __construct(AbstractEmail $email, \PDO $db)
+    {
+        $this->email = $email;
+
+        $stmt = $db->prepare('SELECT `id`, `email` FROM `users` WHERE `email` = :email;');
+        $stmt->execute([':email' => $this->email]);
+        $count = $stmt->rowCount();
+
+        if (!$count) {
```

Comment:
Rather that checking if email is already in database, couldn't you just use REPLACE INTO? Which would overwrite existing emails as themselves, and create new entries for new emails.

Reply:
replace into does not work in all database managers. :)